### PR TITLE
Fix local import error, and support customized name on extension package

### PR DESCRIPTION
**Pull request by squirrelsc**
See destails in commits.

### CODE REVIEW

File: lisa/util/package.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import sys
+from pathlib import Path
+from typing import Optional
+
+from lisa.util.logger import Logger, get_logger
+
+"""
+Reasons to import packages in LISA:
+
+1. Some modules are not be imported by Python automatically. So it need to search and
```
Comment (juhlee-microsoft) [inline lisa/util/package.py]: Wording - 'are not imported'

Comment (juhlee-microsoft) [reply lisa/util/package.py]: 'it needs to'

File: lisa/schema.py [hunk 2]
```diff
@@ -207,14 +207,43 @@ class Parent:
...
+    path: str
+    name: Optional[str] = None
+
+    @classmethod
+    def from_raw(cls, raw_data: Any) -> List["Extension"]:
+        results: List[Extension] = []
+
+        if isinstance(raw_data, Dict):
+            # for compatibility, convert extension to list of strings
+            extension_v1: ExtensionV1 = ExtensionV1.schema().load(  # type: ignore
```
Comment (juhlee-microsoft) [inline lisa/schema.py:232]: What does the comment mean in the line 230 and 241?

File: lisa/schema.py [hunk 3]
```diff
@@ -207,14 +207,43 @@ class Parent:
 
 @dataclass_json()
 @dataclass
-class Extension:
+class ExtensionV1:
```
Comment (glima) [inline lisa/schema.py:210]: Why can't we fix all the users and get away with legacy stuff here?

Comment (squirrelsc) [reply lisa/schema.py:210]: It's fixing, there are some used in critical pipelines. I will double check if all of them done.

**Comment by glima** [comment]
Now I get it why you need to import modules on LISA itself: you make use of type.__subclasses__(). Please document that near such usage, to clarify

**Comment by squirrelsc** [comment]
> Now I get it why you need to import modules on LISA itself: you make use of type.**subclasses**(). Please document that near such usage, to clarify

Will add it.

File: lisa/util/package.py (hunk 1 above)
Comment (squirrelsc) [reply lisa/util/package.py]: Thank you, Ju!

File: lisa/schema.py (hunk 2 above)
Comment (squirrelsc) [reply lisa/schema.py:232]: improved

**Pull request merged**